File: examples/java/SynthTerminal.java
// A runnable Java example: drive a synthetic feed and print a frame.
//
//   cargo build -p wickra-terminal-c
//   mvn -f bindings/java/pom.xml -q package -DskipTests
//   javac -cp bindings/java/target/classes examples/java/SynthTerminal.java -d examples/java/out
//   java --enable-native-access=ALL-UNNAMED \
//        -Dnative.lib.dir=target/debug \
//        -cp "bindings/java/target/classes;examples/java/out" SynthTerminal
import org.wickra.terminal.Terminal;

public final class SynthTerminal {
    private static final String CONFIG =
            "{\"sources\":[{\"Synth\":{\"seed\":1}}],"
                    + "\"layout\":{\"panels\":[{\"kind\":\"Chart\",\"rect\":{\"x\":0,\"y\":0,\"w\":100,\"h\":100}}]}}";

    public static void main(String[] args) {
        try (Terminal term = new Terminal(CONFIG)) {
            term.command("{\"type\":\"Subscribe\",\"source\":0,\"symbol\":\"BTC/USDT\"}");
            String raw = "";
            for (int i = 0; i < 20; i++) {
                raw = term.command("{\"type\":\"Tick\"}");
            }
            System.out.println("wickra-terminal " + Terminal.version());
            System.out.println(raw);
        }
    }
}

File: examples/go/synth_terminal.go
// A runnable Go example: drive a synthetic feed and print a frame.
//
//	cargo build --release -p wickra-terminal-c
//	# stage the library under bindings/go/lib/<goos>_<goarch>/ (CI does this)
//	cd examples/go && go run .
package main

import (
	"fmt"

	wickra "github.com/wickra-lib/wickra-terminal-go"
)

const config = `{"sources":[{"Synth":{"seed":1}}],` +
	`"layout":{"panels":[{"kind":"Chart","rect":{"x":0,"y":0,"w":100,"h":100}}]}}`

func main() {
	term, err := wickra.New(config)
	if err != nil {
		panic(err)
	}
	defer term.Close()

	if _, err := term.Command(`{"type":"Subscribe","source":0,"symbol":"BTC/USDT"}`); err != nil {
		panic(err)
	}
	var raw string
	for i := 0; i < 20; i++ {
		raw, err = term.Command(`{"type":"Tick"}`)
		if err != nil {
			panic(err)
		}
	}

	fmt.Println("wickra-terminal", wickra.Version())
	fmt.Println(raw)
}
